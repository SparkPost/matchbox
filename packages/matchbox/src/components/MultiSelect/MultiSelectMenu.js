import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { ActionList } from '../ActionList';
import styles from './MultiSelectMenu.module.scss';

function MultiSelectMenu(props) {
  const { items, menuRef, isOpen, maxHeight, ...rest } = props;

  const listClasses = classnames(
    styles.List,
    isOpen && styles.open
  );

  return (
    <div {...rest} ref={menuRef} className={listClasses}>
      <ActionList actions={items} maxHeight={maxHeight} />
    </div>
  );
}

MultiSelectMenu.propTypes = {
  /**
   * Array of items. see ActionList prop types.
   */
  items: PropTypes.array,
  /**
   * Controls menu visibility
   */
  isOpen: PropTypes.bool,
  /**
   * Maps to Downshift's refKey set in getMenuProps. refKey must be set to "menuRef"
   */
  menuRef: PropTypes.func
};

MultiSelectMenu.defaultProps = {
  items: []
};

export default MultiSelectMenu;
