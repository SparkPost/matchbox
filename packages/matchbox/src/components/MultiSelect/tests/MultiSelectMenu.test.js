import React from 'react';
import MultiSelectMenu from '../MultiSelectMenu';
import { mount } from 'enzyme';

describe('MultiSelectMenu', () => {
  const subject = (props = {}) => mount(<MultiSelectMenu {...props}/>);
  const items = [
    { content: 'foo' },
    { content: <div>bar</div> }
  ];

  it('should render items correctly', () => {
    expect(subject({ items })).toMatchSnapshot();
  });

  it('should render correctly when open', () => {
    const menu = subject({ items, isOpen: true });
    expect(menu.find('.List')).toHaveAttributeValue('class', 'List open');
  });
});
