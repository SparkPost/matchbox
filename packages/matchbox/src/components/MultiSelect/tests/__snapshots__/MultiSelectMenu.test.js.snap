// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectMenu should render items correctly 1`] = `
<MultiSelectMenu
  items={
    Array [
      Object {
        "content": "foo",
      },
      Object {
        "content": <div>
          bar
        </div>,
      },
    ]
  }
>
  <div
    className="List"
  >
    <ActionList
      actions={
        Array [
          Object {
            "content": "foo",
          },
          Object {
            "content": <div>
              bar
            </div>,
          },
        ]
      }
      groupByKey="section"
    >
      <div
        className="ActionList"
        style={
          Object {
            "maxHeight": "none",
          }
        }
      >
        <Section
          key="0"
          section={
            Array [
              Object {
                "content": "foo",
              },
              Object {
                "content": <div>
                  bar
                </div>,
              },
            ]
          }
        >
          <div
            className="Section"
          >
            <UnstyledLink
              className="Action"
              key="0"
            >
              <a
                className="Action"
              >
                foo
              </a>
            </UnstyledLink>
            <UnstyledLink
              className="Action"
              key="1"
            >
              <a
                className="Action"
              >
                <div>
                  bar
                </div>
              </a>
            </UnstyledLink>
          </div>
        </Section>
      </div>
    </ActionList>
  </div>
</MultiSelectMenu>
`;
